great_expectations/metrics/metric.py

```diff
@@ -120,16 +122,20 @@ def __new__(cls, *args, **kwargs):
             raise AbstractClassInstantiationError(cls.__name__)
         return super().__new__(cls)
 
+    def metric_id_for_batch(self, batch_id: str) -> MetricConfigurationID:
+        return self.config(batch_id=batch_id).id
+
     def config(self, batch_id: str) -> MetricConfiguration:
         # This class is frozen so Metric._to_config will always return the same value
         # when the same batch_id is passed in.
         if not batch_id:
             raise EmptyStrError("batch_id")
+
         config = Metric._to_config(
             instance_class=self.__class__,
+            batch_id=batch_id,
```

Comment:
We compute the config id in the `__init__` method so it is invariant. That means when I set the batch_id after instantiation, it didn't update the id. This was a bug since that means different batches would have the same id.